test/e2e/helm/kuma_helm_deploy_global_zone.go

```diff
@@ -122,4 +130,27 @@ func ZoneAndGlobalWithHelmChart() {
 			g.Expect(err).ToNot(HaveOccurred())
 		}, "30s", "1s").Should(Succeed())
 	})
+
+	Context("Intercommunication CP server catalog on Global CP", func() {
+		countInstancesInCatalog := func(g Gomega) int {
+			out, err := k8s.RunKubectlAndGetOutputE(c1.GetTesting(), c1.GetKubectlOptions(Config.KumaNamespace), "get", "configmap", "cp-catalog", "-o", "jsonpath={.data.config}")
+			g.Expect(err).ToNot(HaveOccurred())
+
+			instances := catalog.Instances{}
+			g.Expect(json.Unmarshal([]byte(out), &instances)).To(Succeed())
+			return len(instances.Instances)
+		}
+
+		It("should update instances in catalog when we scale CP", func() {
+			// given
+			Eventually(countInstancesInCatalog, "30s", "1s").Should(Equal(2))
+
+			// when
+			_, err := k8s.RunKubectlAndGetOutputE(c1.GetTesting(), c1.GetKubectlOptions(Config.KumaNamespace), "scale", "deployment", Config.KumaServiceName, "--replicas", "1")
+
+			// then
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(countInstancesInCatalog, "30s", "1s").Should(Equal(1))
+		})
```

Comment:
Could you do a full rolling restart with 2 instances and check there's still the 2 instances in the catalog?

Reply:
Do you mean instead of scaling down or in addition to scaling down?

Reply:
I mean do: `kubectl rollout restart deployment kuma-control-plane`

Reply:
refactored